### PR TITLE
Add a separate media type for 4k/UHD

### PR DESCRIPTION
**Pull request by gjnoonan**
As requested on the forums, as well as scratching my own itch. 4K releases are now treated as separate media-type when imported. This allows the user to not only; see visually which releases are 4k,if they have format icons enabled within their config, but also filter to 4K releases.

Works nicely on a test DB I put together.

Screenshot 2020-01-05 at 06 50 19

Screenshot 2020-01-05 at 06 51 00

Screenshot 2020-01-05 at 06 51 29

Still left to do is the movie picker/chooser, but this won't be too much work for a later commit.

_More information in the commit messages, I would suggest not squashing commits when you merge._

### CODE REVIEW

**Comment by julmud** [comment]
Wow, thanks a lot for this Pull Request! I'll have a look at it and merge it as soon as I can find the time to do it, but it probably won't be today nor tomorrow, alas.

**Comment by gjnoonan** [comment]
No worries! I have a few more lined up to send in once I finish testing them, and splitting the hunks--but none are dependent on this being in master :-)

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+/vendor/
+ERROR_LOG
```
Comment (julmud) [inline .gitignore:1]: Thanks. I wasn't sure whether or not it was a good idea to commit the file, but it makes sense to ignore it at least. We should probably also add the DEBUG_LOG file to the ignore list. That one should never be committed, but it can be created when developers are trying to understand what happens in the code. ;-)
```suggestion
ERROR_LOG
DEBUG_LOG
```

File: globalinits.php [hunk 2]
```diff
@@ -63,7 +66,11 @@
 	MEDIA_TYPE_HDDVD	=> array('FormatIcon' => 'gfx/iconhddvdpng.png',  'Icon' => 'gfx/hddvd.jpg',  'Banner' => 'gfx/Banner_HDDVD.png'),
 	MEDIA_TYPE_HDDVD_DVD	=> array('FormatIcon' => 'gfx/iconhddvdpng.png',  'Icon' => 'gfx/hddvd.jpg',  'Banner' => 'gfx/Banner_HDDVD_DVD.png'),
 	MEDIA_TYPE_BLURAY	=> array('FormatIcon' => 'gfx/iconbluraypng.png', 'Icon' => 'gfx/bluray.jpg', 'Banner' => 'gfx/Banner_BluRay.png'),
-	MEDIA_TYPE_BLURAY_DVD	=> array('FormatIcon' => 'gfx/iconbluraypng.png', 'Icon' => 'gfx/bluray.jpg', 'Banner' => 'gfx/Banner_BluRayDVD.png')
+	MEDIA_TYPE_BLURAY_DVD	=> array('FormatIcon' => 'gfx/iconbluraypng.png', 'Icon' => 'gfx/bluray.jpg', 'Banner' => 'gfx/Banner_BluRayDVD.png'),
+	MEDIA_TYPE_ULTRAHD	=> array('FormatIcon' => 'gfx/iconultrahd.png', 'Icon' => 'gfx/ultrahd.jpg', 'Banner' => 'gfx/Banner_UltraHD.png'),
+	MEDIA_TYPE_ULTRAHD_BLURAY	=> array('FormatIcon' => 'gfx/iconultrahd.png', 'Icon' => 'gfx/ultrahd.jpg', 'Banner' => 'gfx/Banner_UltraHDBD.png'),
```
Comment (julmud) [inline globalinits.php:71]: Small alignment problem
```suggestion
	MEDIA_TYPE_ULTRAHD_BLURAY    => array('FormatIcon' => 'gfx/iconultrahd.png', 'Icon' => 'gfx/ultrahd.jpg', 'Banner' => 'gfx/Banner_UltraHDBD.png'),
```

File: incupdate.php [hunk 3]
```diff
@@ -476,12 +476,20 @@ function InitialiseCommonTable($which) {
 	}
 }
 
-function FigureOutBuiltinMediaType($mediatypedvd, $mediatypehddvd, $mediatypebluray) {
+function FigureOutBuiltinMediaType($mediatypedvd, $mediatypehddvd, $mediatypebluray,$mediatypeultrahd) {
```
Comment (julmud) [inline incupdate.php]: Nitpicking: 
```suggestion
function FigureOutBuiltinMediaType($mediatypedvd, $mediatypehddvd, $mediatypebluray, $mediatypeultrahd) {
```

**Review by julmud — COMMENTED** [review]
I still want to do a few tests later today or tomorrow evening before merging, but it looks good. Thanks a lot!

File: globalinits.php (hunk 2 above)
Comment (gjnoonan) [reply globalinits.php:71]: Good spot thanks, I thought I caught them all when I went back through.

**Comment by gjnoonan** [comment]
> I still want to do a few tests later today or tomorrow evening before merging, but it looks good. Thanks a lot!

Thanks @julmud I will re-roll with the above changes today.

File: .gitignore (hunk 1 above)
Comment (gjnoonan) [reply .gitignore:1]: I have now pulled this out into it's own commit 053d2973887aeb406fef28740da4ba674cc1864b

_edit: the commit hash changed, as I edited the commit message and pushed it back up_

**Comment by gjnoonan** [comment]
@julmud I have pushed up V2 of the changes incorporating a suggested change, and introducing a new commit 02b7a5e3e98d4d3eb84ad7ec2b65e73ef0ff01b2 which will add a .gitignore which has your suggested change also.

File: incupdate.php (hunk 3 above)
Comment (gjnoonan) [reply incupdate.php]: resolved in v2

File: globalinits.php (hunk 2 above)
Comment (gjnoonan) [reply globalinits.php:71]: I have not resolved this on with the suggestion, as it still does not look right. The array keys have become too long I think. I will push a patch with a larger change clearing up the array itself.

**Review by julmud — APPROVED** [review]
I've realised tonight that there're a few more files we should add to .gitignore, but I'll leave it for another PR. Thanks a lot!
